gammapy/stats/variability.py

```diff
@@ -346,7 +349,7 @@ def TimmerKonig_lightcurve_simulator(
 
     random_state = get_random_state(random_state)
 
-    frequencies = np.fft.fftfreq(npoints, spacing.value)
+    frequencies = np.fft.fftfreq(int(npoints * leakage_protection), spacing.value)
```

Comment:
maybe define `npoints_extended= int(npoints * leakage_protection)` before,
check that `if leakage_protection!=1: npoints_extended>npoints`, 
and if not send a warning or force `npoints_extended = npoints+1`